feat: add option to return in order in client and streamer (#5404)
jina-ai/jina: docs/fundamentals/client/client.md

````

> A Client connects to a Flow that processes Documents in an asynchronous and very distributed way. This means that the order of the Flow processing the requests may not be the same order as the Client sending the requests.

Reviewer: Is there a way to pass this parameter from a 3rd party client?

Author: No, this is gonna be ready in the `streamer` for the CustomGateway. We can add it later I would say

Author: Mostly external clients do not have this `streaming` capability so easily implemented, so I think is not so so important

> A Client is used to connect to a Flow who processes Documents in an asynchronous and very distributed way. This means that the order in which the Requests are processed may not be the exact same order as they are sent by the Client.

Reviewer 2: ```suggestion
A Client connects to a Flow that processes Documents in an asynchronous and very distributed way. This means that the order of the Flow processing the requests may not be the same order as the Client sending the requests.
```
I think a short example would be useful here. Parsing that text was not simple.

Wait, when we say "who processes Documents in an asynchronous and very distributed way" does that refer to client or flow?

Maybe just rewrite this bit. There's ambiguity

Author: the Flow processes it in an asynchronous and distributed way

Author: There is a test already

> However, you can force the order of the results to be deterministic and the same as they enter the Flow by passing `results_in_order` parameter to {meth}`~jina.clients.mixin.PostMixin.post`.

Reviewer 2: ```suggestion
However, you can force the order of the results to be deterministic and the same as when they enter the Flow by passing `results_in_order` parameter to {meth}`~jina.clients.mixin.PostMixin.post`.
```
